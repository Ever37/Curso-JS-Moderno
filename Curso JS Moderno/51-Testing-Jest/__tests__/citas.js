class CitasÂ {
    constructor() {
        this.citas = []
    }
    agregarCita(cita) {
        this.citas = [...this.citas, cita];
    }
    editarCita(citaActualizada) {
        this.citas = this.citas.map( cita => cita.id === citaActualizada.id ? citaActualizada : cita)
    }

    eliminarCita(id) {
        this.citas = this.citas.filter( cita => cita.id !== id);
    }
}

describe('Probar la clase de Citas', () => {

    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Juan',
            telefono: '1239132',
            fecha: '10-12-2021',
            hora:'10:30',
            sintomas: 'covid'
        };

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });

    test('Actualizar una cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'Juan',
            telefono: '1239132',
            fecha: '10-12-2021',
            hora:'10:30',
            sintomas: 'covid'
        };

        citas.editarCita(citaActualizada);

        // Prueba
        expect(citas).toMatchSnapshot();
    });

    test('Eliminar Cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    });
})