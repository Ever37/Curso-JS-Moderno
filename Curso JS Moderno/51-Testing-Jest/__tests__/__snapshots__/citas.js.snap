// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar una cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1624918896082,
      "mascota": "Nuevo nombre",
      "propietario": "Juan",
      "sintomas": "covid",
      "telefono": "1239132",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1624918896082,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "covid",
      "telefono": "1239132",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;
